src/plugins/chart_expressions/expression_heatmap/public/compoments/heatmap_component.tsx

```diff
@@ -159,17 +212,23 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = ({
       };
     });
   }
-
-  const xAxisMeta = xAxisColumn.meta;
-  const isTimeBasedSwimLane = xAxisMeta.type === 'date';
+  const { min, max } = minMaxByColumnId[valueAccessor!];
+  // formatters
+  const xAxisMeta = xAxisColumn?.meta;
+  const xValuesFormatter = formatFactory(xAxisMeta?.params);
+  const metricFormatter = formatFactory(
+    typeof args.valueAccessor === 'string' ? valueColumn.meta.params : args?.valueAccessor?.format
```

Comment:
nitpick:

```suggestion
    typeof valueAccessor === 'string' ? valueColumn.meta.params : args?.valueAccessor?.format
```

Reply:
I need the args.valueAceessor here to make this check because it can be either a string or a vis_dimension. In case of vis_dimension I am taking the meta params from it.